traveler-swift-core/TravelerKitTests/Mocking/MockNetwork.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  MockNetwork.swift
+//  TravelerKitTests
+//
+//  Created by Omar Padierna on 2019-10-25.
+//  Copyright © 2019 Ata Namvari. All rights reserved.
+//
+
+import Foundation
+
+/// The URLProtocol subclass allowing to intercept the network communication
+/// and provide custom mock responses for the given URLs.
+class MockNetwork: URLProtocol {
+
+  /// The key-pairs of URLs this URLProtocol intercepts with their simulated response.
+  static var mockResponses: [URL: Result<Data, Error>] = [:]
```

Comment:
why static?

Reply:
This renders a singleton pattern which I dont see why we need

Reply:
One of the reasons is because the `canInit` function is a class function which needs to check for the available keys from the `mockResponses` object.  This can be avoided by having a separate set of paths to check against.  However, another problem exists, the instance of `MockNetwork` needs to stay in memory through the duration of the tests.  This can be remedied by using class instances in `MockAPI.swift` for `MockNetwork` and `PassengerKitTests.swift` for `MockAPI`.  This, however, results in unexpected behaviour in which the `mockResponses` data is not populated prior to the invocation of `startLoading`.  For reference on this implementation, please refer to this article: https://vojtastavik.com/2019/09/12/mocking-network-calls-using-urlprotocol/